### PR TITLE
[ZFE-16] Checkout Page

### PR DESCRIPTION
**Pull request by zakrzaq**

### CODE REVIEW

File: components/Base/DatePicker.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    :for="name"
+    class="inline-block min-w-[250px]"
+  >
+    <slot class="block mb-7" />
+    <input
+      type="date"
+      id="`date-picker__${name}`"
+      :name="name"
+      :value="new Date().toJSON().slice(0, 10)"
+      @change="$emit('update:date', ($event.target as HTMLInputElement).value)"
```
Comment (E1mir) [inline components/Base/DatePicker.vue:12]: refactor value as data, and move change method to methods

File: components/Base/CounterButton.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import MinusIcon from '~/assets/icons/minus.svg?skipsvgo';
+
+const props = defineProps<{
+  counter: number;
+  type: 'plus' | 'minus';
+}>();
+
+const emit = defineEmits(['update:counter']);
+
+const isPlusButton = computed(() => (props.type === 'plus' ? true : false));
```
Comment (E1mir) [inline components/Base/CounterButton.vue:22]: `(() => props.type === 'plus')`

File: pages/checkout/index.vue [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <h1 class="mb-11">Checkout</h1>
+  <div class="grid grid-cols-2 gap-x-72 mb-80">
+    <CheckoutDetails />
+    <CheckoutSummary />
+  </div>
+</template>
+
+<script setup lang="ts"></script>
```
Comment (E1mir) [inline pages/checkout/index.vue:9]: I'm getting this warning on this page:

image

**Review by E1mir — CHANGES_REQUESTED** [review]

File: pages/checkout/index.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    formData.days &&
+    formData.license &&
+    formData.insurance
+  );
+});
+
+const handleSubmit = (e: Event) => {
+  e.preventDefault();
+  formValid.value
+    ? console.log('order', formData)
```
Comment (zakrzaq) [inline pages/checkout/index.vue:57]: TO BE REPLACED

File: components/Base/FileUpload.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+const file = ref();
+
+const handleChange = (e: Event) => {
+  e.preventDefault();
+  const target = <HTMLInputElement>e.target;
+  if (target.files && target.files[0]) {
+    const incoming: File = target.files[0];
+    file.value = incoming;
+    emit('update:file', file.value);
```
[the remark this answers is not in the document]
Comment (zakrzaq) [reply components/Base/FileUpload.vue:44]: Just felt more natural to me. For some reason I am not hot on using html refs unless really needed

[the remark this answers is not in the document]
Comment (zakrzaq) [reply components/Base/FileUpload.vue]: Ended up using a ref 😶‍🌫️

[the remark this answers is not in the document]
Comment (WawrzenskiPatryk) [reply components/Base/FileUpload.vue]: Also, I think you can just use `prevent` modifier on the `change` event so you don't have to call `e.preventDefault()` in your function.
```js
@change.prevent="handleChange"
```

Check it here: https://vuejs.org/guide/essentials/event-handling.html#event-modifiers

File: pages/checkout/index.vue [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        variant="text"
+      >
+        Back
+      </BaseButton>
+      <BaseButton
+        arrow="right"
+        :disabled="!formValid"
+        @click="handleSubmit"
+        >Book</BaseButton
+      >
```
Comment (WawrzenskiPatryk) [inline pages/checkout/index.vue:24]: Formatting 😄

File: components/Base/RadioOption.vue [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    </div>
+    <div class="flex flex-col px-5 py-3">
+      <p class="text-sm font-bold">{{ option.title }}</p>
+      <p class="text-xs">{{ option.description }}</p>
+    </div>
+    <span
+      v-if="option.detail"
+      class="absolute top-3 right-5 text-xs font-bold"
+      >${{ option.detail }}</span
+    >
```
Comment (WawrzenskiPatryk) [inline components/Base/RadioOption.vue:23]: Formatting 😄

**Review by WawrzenskiPatryk — CHANGES_REQUESTED** [review]

File: components/Base/CounterButton.vue [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+const isPlusButton = computed(() => props.type === 'plus');
+
+const handleClick = () => {
+  isPlusButton.value ? incrementCounter() : decrementCounter();
+};
+
+const incrementCounter = () => emit('update:counter', props.counter + 1);
+const decrementCounter = () =>
+  emit('update:counter', props.counter > 0 ? props.counter - 1 : props.counter);
```
Comment (elmir-ismayilzada) [inline components/Base/CounterButton.vue:30]: `>= 0`, we can set 0 days

image

File: components/Base/FileUpload.vue [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      @drop.prevent="handleDrop"
+      @change="handleChange"
+    >
+      <slot v-if="!selectedFile" />
+      <p
+        v-else
+        class="text-xs"
+      >
+        {{ selectedFile.name }}
+      </p>
```
Comment (elmir-ismayilzada) [inline components/Base/FileUpload.vue:20]: Here, as you see I can add `.torrent` file, but I shouldn't. 

Also add padding :)

image

File: components/Base/RadioGroup.vue [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    </div>
+  </RadioGroup>
+</template>
+
+<script setup lang="ts">
+import { RadioGroup, RadioGroupLabel, RadioGroupOption } from '@headlessui/vue';
+import { RadioOption } from '~/types/radio-options';
+
+defineProps<{
+  option: RadioOption;
```
Comment (elmir-ismayilzada) [inline components/Base/RadioGroup.vue:31]: I think the name should be `optionValue` or `value` or may be `selected`, WDYT? 🤔

File: components/Base/FileUpload.vue [hunk 11]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div>
+    <label
+      class="flex flex-col items-center justify-center w-full bg-gray-100 border-[2px] border-dashed h-20 rounded-3xl"
+      :accept="accept"
```
Comment (elmir-ismayilzada) [inline components/Base/FileUpload.vue:5]: I'm not sure that accept should be here 😅 For example I can drag & drop `.torrent` file, see below

File: components/Checkout/Summary.vue [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+const reservationCost = computed(() => {
+  return props.order.days && product.value && product.value.price
+    ? props.order.days * +product.value.price
+    : 0;
+});
+
+const total = computed(() => {
+  return props.order.insurance?.detail
+    ? reservationCost.value + parseInt(props.order.insurance.detail)
```
Comment (elmir-ismayilzada) [inline components/Checkout/Summary.vue:65]: `parseInt` will parse only integer numbers, it will "cut" decimals.

**Review by elmir-ismayilzada — CHANGES_REQUESTED** [review]

File: components/Base/FileUpload.vue (hunk 11 above)
Comment (zakrzaq) [reply components/Base/FileUpload.vue:5]: This is generic file upload, so accepts all files. The issue is I have not bound a prop in next component `License upload` that should have limit it to document and images

File: components/Checkout/Summary.vue (hunk 12 above)
Comment (zakrzaq) [reply components/Checkout/Summary.vue:65]: That is intended behavior as detail is a price.

File: components/Base/RadioGroup.vue (hunk 10 above)
Comment (zakrzaq) [reply components/Base/RadioGroup.vue:31]: `selected` sounds good to me :)

**Comment by elmir-ismayilzada** [comment]
@zakrzaq You have a conflicts

**Review by elmir-ismayilzada — APPROVED** [review]
I went through it again, no issues spotted. Good work!

**Review by WawrzenskiPatryk — APPROVED** [review]
LGTM :)

**Pull request merged**